UltraStar Play/Assets/Common/Audio/AudioUtils.cs

```diff
@@ -15,43 +18,41 @@ public static AudioClip GetAudioClip(string path)
         }
         string fileExtension = System.IO.Path.GetExtension(path);
 
-        using (new DisposableStopwatch("Loaded audio in <millis> ms"))
+        if (fileExtension.ToLowerInvariant().Equals(".mp3"))
         {
-            if (fileExtension.ToLowerInvariant().Equals(".mp3"))
-            {
-                return LoadMp3(path);
-            }
-            else
-            {
-                return LoadAudio(path);
-            }
+            AudioClip audioClip = LoadMp3(path);
+            return audioClip;
+        }
+        else
+        {
+            return LoadAudio(path, streamAudio);
         }
     }
 
-    private static AudioClip LoadAudio(string path)
+    private static AudioClip LoadAudio(string path, bool streamAudio)
     {
-        using (UnityWebRequest www = UnityWebRequestMultimedia.GetAudioClip("file://" + path, AudioType.UNKNOWN))
+        using (UnityWebRequest webRequest = UnityWebRequestMultimedia.GetAudioClip("file://" + path, AudioType.UNKNOWN))
         {
-            www.SendWebRequest();
+            DownloadHandlerAudioClip downloadHandler = webRequest.downloadHandler as DownloadHandlerAudioClip;
+            downloadHandler.streamAudio = streamAudio;
 
-            if (www.isNetworkError)
+            webRequest.SendWebRequest();
+            if (webRequest.isNetworkError || webRequest.isHttpError)
             {
                 Debug.LogError("Error Loading Audio: " + path);
-                Debug.LogError(www.error);
+                Debug.LogError(webRequest.error);
                 return null;
             }
-            else
+
+            while (!webRequest.isDone)
             {
-                while (!www.isDone)
-                {
-                    Task.Delay(30);
-                }
-                return DownloadHandlerAudioClip.GetContent(www);
+                Task.Delay(30);
             }
+            return downloadHandler.audioClip;
         }
     }
 
-    private static AudioClip LoadMp3(string path)
+    public static AudioClip LoadMp3(string path)
```

Comment:
why did you make LoadMp3 public? It is not used anywhere else as far as I can see.

Reply:
I needed multiple attemps to find a working solution for async loading of audio. The Unity documentation about streamAudio is not so clear. Anyway, I refactored back and forth between AudioUtils and AudioManager. Looks like I forgot to change this back to private.

Will do with next PR.